File: sort/insertionSort/insertionSort.js
// function sort(arr) {
//   const swap = (arr, i, j) => ([arr[i], arr[j]] = [arr[j], arr[i]]);

//   for (let i = 1; i < arr.length; i++){
//     for (let j = i - 1; j >= 0; j--) {
//       if (arr[j+1] < arr[j]) {
//         swap(arr, j, j + 1);
//       } else {
//         break;
//       }
//     }
//   }
//   return arr;
// }

function sort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let currVal = arr[i]
    for (var j = i - 1; j >= 0 && arr[j] > currVal; j--) {
        arr[j+1] = arr[j]
    }
    arr[j+1] = currVal
  }
  return arr;
}

File: sort/insertionSort/insertion_sort.rb
def insertion_sort(arr)
    (1...arr.size).each do |i|
        curr_val = arr[i]
        (i-1).downto(0) do |j|
            if curr_val < arr[j]
                arr[j+1] = arr[j]
                arr[j] = curr_val
            end
        end
    end
    return arr
end
